frontend/components/Main/MainHeader/AgentButton/index.tsx

```diff
@@ -0,0 +1,273 @@
+    }
+
+    // Can assume successful deployment
+    // resume polling, optimistically update service status (poll will update, if needed)
+    setIsServicePollingPaused(false);
+    setIsBalancePollingPaused(false);
+    setServiceStatus(DeploymentStatus.DEPLOYED);
+  }, [
+    wallets,
+    setIsServicePollingPaused,
+    setIsBalancePollingPaused,
+    setServiceStatus,
+    masterSafeAddress,
+    showNotification,
+    serviceTemplate,
+    service,
+  ]);
+
+  const isDeployable = useMemo(() => {
+    if (serviceStatus === DeploymentStatus.DEPLOYED) return false;
+    if (serviceStatus === DeploymentStatus.DEPLOYING) return false;
+    if (serviceStatus === DeploymentStatus.STOPPING) return false;
+
+    // case where service exists & user has initial funded
+    if (service && storeState?.isInitialFunded) {
+      if (!safeOlasBalanceWithStaked) return false;
+      // at present agent will always require staked/bonded OLAS (or the ability to stake/bond)
+      return safeOlasBalanceWithStaked >= requiredOlas;
+    }
+
+    // case if agent is evicted and user has met the staking criteria
+    if (isEligibleForStaking && isAgentEvicted) return true;
+
+    const hasEnoughOlas = (safeOlasBalanceWithStaked ?? 0) >= requiredOlas;
+    const hasEnoughEth = (totalEthBalance ?? 0) > requiredGas;
+
+    return hasEnoughOlas && hasEnoughEth;
+  }, [
+    isAgentEvicted,
+    isEligibleForStaking,
+    safeOlasBalanceWithStaked,
+    service,
+    serviceStatus,
+    storeState?.isInitialFunded,
+    totalEthBalance,
+  ]);
+
+  const buttonProps: ButtonProps = {
+    type: 'primary',
+    size: 'large',
+    disabled: !isDeployable,
+    onClick: isDeployable ? handleStart : undefined,
+  };
+
+  const buttonText = `Start agent ${service ? '' : '& stake'}`;
+
+  return <Button {...buttonProps}>{buttonText}</Button>;
+};
+
+export const AgentButton = () => {
+  const { service, serviceStatus, hasInitialLoaded } = useServices();
+  const { isEligibleForStaking, isAgentEvicted } = useStakingContractInfo();
+
+  return useMemo(() => {
+    if (!hasInitialLoaded) {
+      return <Button type="primary" size="large" disabled loading />;
+    }
+
+    if (serviceStatus === DeploymentStatus.STOPPING) {
+      return <AgentStoppingButton />;
+    }
+
+    if (serviceStatus === DeploymentStatus.DEPLOYING) {
+      return <AgentStartingButton />;
+    }
+
+    if (serviceStatus === DeploymentStatus.DEPLOYED) {
+      return <AgentRunningButton />;
+    }
+
+    if (!isEligibleForStaking && isAgentEvicted) return <CannotStartAgent />;
+
+    if (
+      !service ||
+      serviceStatus === DeploymentStatus.STOPPED ||
+      serviceStatus === DeploymentStatus.CREATED ||
+      serviceStatus === DeploymentStatus.BUILT ||
+      serviceStatus === DeploymentStatus.DELETED
+    ) {
+      return <AgentNotRunningButton />;
+    }
+
+    return (
+      <Button type="primary" size="large" disabled>
```

Comment:
maybe open discord on click, wdyt? :D

Reply:
great suggestion, will raise in pearl slack channel